### PR TITLE
Balance: Rebalance of ship mass and handling across the board

### LINKED ISSUE #5271: There is a lack of progression in speed related to ship sizes
**Is your feature request related to a problem? Please describe.**
Currently in game there is a progression as you increase the size of your ship. Generally, the bigger the ship within a tier, the more HP it has, the more DPS it can output, and the more outfit/weapon/engine space it has. One would also think though that the bigger your ship, the slower it goes. While it is roughly the case that larger ships are slower than smaller ships, it is very easy to equip a larger ship with engines that make it as fast as or faster than a much smaller ship. Therefore, there is currently no real progression in speed related to ship size. Increasing the size of your ship yields no downsides, as the one expected downside of lower speed does not exist.

**Related Issue Links**
Other issues that either speak directly to or are related to the balance between large and small ships:
#1932 was closed because turrets were given turn rates. While this helped to give smaller ships a chance, the issue of speed still exists.
#4034 The issue with smaller ships vs larger ships is related to the issue of fighters being underwhelming at the moment.
#4407 suggested using multipliers to make generators that are more attractive to smaller ships, but I suggested that we instead simply make smaller ships regenerate faster. (This is also something mentioned in #1932.)
#4697 discusses ways to make smaller ships have more of a chance against larger ships. 

**Describe the solution you'd like**
Rebalance the mass and drag of ships to have a much more drastic jump between categories of ships.

For example, right now the difference in the average mass of a human interceptor versus a human light warship is only +80%. From light to medium warship it's +87%, and from medium to heavy warship it's +126%.
In terms of drag, the progression is +127% from interceptor to light warship, +86% from light warship to medium warship, and +57% from medium warship to heavy warship.

By increasing the jump in mass and drag as you go up a ship category, we'd make it so that larger ships handle noticeably slower than smaller ships, being the trade off for more HP and DPS.

**Describe alternatives you've considered**
Another point of concern might be the progression in power between various types of engines. A ship with atomic engines will be going MUCH faster than a ship with ion engines of the same size. Perhaps we want to reduce the difference between the slowest and fastest engines. Currently, a tiny atomic thruster has nearly double the thrust per ton as a tiny ion thruster. (87% more to be exact.) This difference in power continues all the way up the size progression of engines. (The huge atomic thruster has 81% more thrust per ton than the huge ion thruster.)

And in terms of progression of power as you increase in engine size, at the same time that the average human heavy warship has 565% more drag and 661% more mass than a human interceptor, huge thrusters have about 1260% more thrust in the case of ion and plasma thrusters, and 807% more thrust in the case of atomic engines when compared to the tiny thruster of the same engine category. 

This is to point out that on average, **the power of human engines as you increase in size currently outpaces the increase in drag and mass.** Therefore, we either need to reduce the power of engines with size or increase the drag and mass more with size.

**Additional context**
Should we go the route of increasing the mass of ships, we need to remember that hit force and optical tracking are currently balanced assuming the average mass of ships of certain categories.

### PR DESCRIPTION
**Pull request by Quantumshark**
**Balance: resolves #5271**

## Summary
All engines' thrust and turn have been doubled (to provide room for small ships to be made faster without their hull masses going negative).
Ships have received multiplicative increases to their net mass (achieved by tweaking hull mass, but taking into consideration that and outfit space), and equally to their drag (top speed to acceleration ratios remain unchanged here, and could be tweaked as desired in future using the acceleration multiplier attribute from #9293).
The numbers were decided by first determining ranges for each category, and then determining where in that range each ship should sit.
The category-based ranges, in no particular order, are as follows:
Fighters and drones: x1 to x1.1
Interceptors: x1.1 to x1.4
Light Warships: x1.4 to x1.8
Medium Warships: x1.8 to x2.2
Heavy Warships: x2.2 to x3
Superheavies: x3 or higher
Light Freighters: x1.4 to x1.8
Heavy Freighters: x1.8 to x2.2
Space Liners: x1.8 to x2.2
Utility: x1.8 to x3 (this is a notably disparate class)
Transport: x1.1 to x1.8
The full list of the multiplier used for each ship is here:
[massrebalance multipliers.txt](https://github.com/endless-sky/endless-sky/files/12895071/massrebalance.multipliers.txt)

Ships with integrated turn and thrust and no engines received x2 to mass, drag, turn and thrust for consistency.

## Reasoning
It has been widely agreed that steeper scaling in handling and mass is something we want, so I'll mostly focus this section on why I think this set of changes is the best solution to that rather than on why that's something that's desired.

Progression of mobility with size should, in my view, mean within categories as well as between them. Approaches that simply apply a modifier based solely on category result in huge steps up between the boundaries of categories, and leave the issue of 'larger ships are always better' within categories. That's why, for example, the Vanguard's mobility is nerfed a lot less in this PR than the Carrier's.

I seek to keep the range of speeds manageable, so that ships at both ends are comfortable to pilot. Inteceptors shouldn't end up with more than 200 turn (fighters and drones may, but are not necessarily designed to be piloted by players), and Heavy Warships still retain enough mobility to be enjoyable to play (some Superheavies which are either unobtainable or practically unobtainable do not, in a similar vein to why fighters and drones can go 'too far' in the opposite direction).

Heavy Freighters received adjustments in line with medium warships rather than with heavy warships. This is because heavy freighters generally have similar outfit space numbers to medium warships, and because heavy freighters definitely do not need to be any slower than they already are.

## Save File
Any, or better yet a new pilot (as old ships won't be updated)

## Todo:

- [x] Adjust hit force and firing force in the same way as thrust and turn
- [x] Adjust heat dissipations to keep max heat constant
- [x] Confirm small ships aren't too fast
- [x] Confirm large ships aren't too slow
- [ ] Test campaign combat balance

### CODE REVIEW

**Comment by Zitchas** [comment]
You *could* just pull straight from Delta, and you'd get the lateral thrust code that some people have been asking for too. ;)

**Comment by Quantumshark** [comment]
These are different numbers to the ones in the Delta fork, and are intended to stand up without requiring changes to the game mechanics. There's room for game mechanics changes too, and potentially for updates to these values informed by those, but I indend for these changes to represent an improvement and a working approach within the current mechanics.

**Comment by Zitchas** [comment]
Oh, definitely. The more variations we try the better. 

My previous comment was somewhat tongue in cheek, as we generally don't do mechanics + data changes in the same PR, which pulling from Delta would definitely do. Just I know that some people have been hoping to get the lateral thrust mechanics PR'd over, which that would have done.

**Comment by bene-dictator** [comment]
So far what I've noticed after ten minutes running the Deep's Behemoths down to Kraz and back:
* Woah, the Corvette's turn is speedy with an A375 Steering (257 turn). I've set auto steer as the same key so I can aim at my enemy, but otherwise it's a lot faster than I can handle.
* Daggers blink across the screen, because they have a max speed of... 3,558 in stock configuration. I'm not sure if they're hitting their targets because they end up overshooting. The Boxwing has a top speed of 1,261 but its thrusters make it look like it should only have something like 400 haha.

**Comment by Quantumshark** [comment]
> So far what I've noticed after ten minutes running the Deep's Behemoths down to Kraz and back:
> 
> * Woah, the Corvette's turn is speedy with an A375 Steering (257 turn). I've set auto steer as the same key so I can aim at my enemy, but otherwise it's a lot faster than I can handle.

The Corvette's mobility should only have improved by about 5% (2/1.9 = 1.053), so if an A375 steering is too much for it in this PR I suspect it'd be too much without this PR too.
The Berserker's stock X3200 steering similarly pushes its turn rate a little too high for most people, and since that's the stock loadout it may need to be looked at.

> * Daggers blink across the screen, because they have a max speed of... 3,558 in stock configuration. I'm not sure if they're hitting their targets because they end up overshooting. The Boxwing has a top speed of 1,261 but its thrusters make it look like it should only have something like 400 haha.

I presume with double the top speed, double the acceleration and double the turning (which is what the Dagger receives here), the AI's control of the ship should be largely or entirely unaltered, since its reaction time isn't limited the way a human's is. The speed increase should also improve their survivability. 
The Boxwing already had a top speed of a little over 600.

**Comment by Saturnalia9** [comment]
> I presume with double the top speed, double the acceleration and double the turning (which is what the Dagger receives here), the AI's control of the ship should be largely or entirely unaltered, since its reaction time isn't limited the way a human's is. The speed increase should also improve their survivability. 
> The Boxwing already had a top speed of a little over 600. Its mobility could be reduced if needed.

It (Dagger) might survive better, but that might also hit its DPS as well.

**Comment by bene-dictator** [comment]
Screenshot 2023-10-18 at 11 05 34 PM

A370 Thrusters (mass 53t) vs Impala Thruster (mass 58t)

Was this similarly placed before due to #8842, or is the A370 particularly inefficient now?

> It (Dagger) might survive better, but that might also hit its DPS as well.

Daggers appear to fly past (overshoot) their targets and spend time having to get back into the action. That obviously brings their DPS down to zero when they are turning around, losing momentum then regaining momentum, then turning around again... like a pendulum.

I'll have to make a video of it.

**Comment by Quantumshark** [comment]
> Screenshot 2023-10-18 at 11 05 34 PM
> 
> A370 Thrusters (mass 53t) vs Impala Thruster (mass 58t)
> 
> Was this similarly placed before due to #8842, or is the A370 particularly inefficient now?

Unless I've made an error somewhere, this PR just doubles every thruster and steering. So, the comparisons between them should be unaffected. 
I suspect this isn't the result of #8842, but rather #6686, which narrowed the gap between atomics and plasma engines considerably.

**Comment by Quantumshark** [comment]
> It (Dagger) might survive better, but that might also hit its DPS as well.

As I understand it, fighters currently are in a good state in terms of overall performance, but are inconvenient for players to use because they lack survivability and must be frequently replaced. From that position, if we just made them more survivable with no tradeoff, they'd be too strong. So, making them more survivable at the expense of damage output might be a good thing.

**Comment by Saturnalia9** [comment]
> > It (Dagger) might survive better, but that might also hit its DPS as well.
> 
> As I understand it, fighters currently are in a good state in terms of overall performance, but are inconvenient for players to use because they lack survivability and must be frequently replaced. From that position, if we just made them more survivable with no tradeoff, they'd be too strong. So, making them more survivable at the expense of damage output might be a good thing.

3500 km/h reduces damage output to almost nothing. 😅

**Comment by Quantumshark** [comment]
Have you tested that? All the testing I've done suggests they're still pretty capable of dealing damage with most weapons, and obviously if equipped with long-range missiles the speed isn't going to make them any less effective.

**Comment by Quantumshark** [comment]
From some combat testing, I've found that with these changes it's generally possible to take out a warship one size category larger than your ship in a 1v1 duel with clever piloting and manoeuvring. A well-piloted interceptor can take down a light warship, a well-piloted light warship can take down a medium warship, and so on.
You still can't take on a heavy warship in an interceptor, but I don't think you'd actually want to even if it were possible - you couldn't capture it, and couldn't fit any plunder that you couldn't get from a smaller ship.

Combat missions involving multiple smaller ships become much more difficult with these changes - I had significant trouble with the Rat Pack and with small marauder fleets when flying a medium warship, which isn't something I've had previously.

Fleets of interceptors are much, *much* more effective for their cost than anything else. 4M credits' worth of Furies (i.e. five of them) with Javelin pods will take down pretty much any single ship, even a heavy warship costing three times that. So, maybe interceptors are overbuffed here? It's something to keep an eye out for.

**Comment by Quantumshark** [comment]
Having played through the first few FW battles with this PR, I've noticed that the ships seem to spread out a lot more than they used to. I suspect the greater variety in top speeds helps a lot with this. So far, no issues with difficulty - if anything, they seem to be slightly easier than before. My guess is that this is because the Gunboat and Rainmaker are quite sluggish for their size, and thus Republic fleets don't stand to benefit as much from the greater mobility of small ships, whereas the interceptor-heavy FW fleets benefit significantly, and possibly also because the Cruiser and Carrier have suffered the greatest losses in mobility.

Travelling through Republic space for Syndicate Diplomacy was still difficult, but not noticeably harder. I was worried that the need to use a ship with lower HP to achieve the mobility that Heavy Warships used to have would make it harder, but in practice large republic fleets are strong enough and good enough at interdicting that HP is irrelevant - if you're being fired on, you're going to be held in place and gunned down, it'll just take longer. So probably a Scout was a better option than a Falcon even before the change simply because it's a smaller target, and with these changes even more so. I've seen people complain that there's currently never any case to use a smaller ship, but I'm now pretty confident that missions like this will provide one.

I've settled on an Engines Marauder Firebird as my flagship, where normally I'd have opted for a Falcon or Vanguard. I've beaten both Falcons and Leviathans 1v1 with it, so I don't really feel any need to try anything bigger.

**Comment by Saturnalia9** [comment]
> You still can't take on a heavy warship in an interceptor, but I don't think you'd actually want to even if it were possible - you couldn't capture it, and couldn't fit any plunder that you couldn't get from a smaller ship. Also, if one interceptor is an even match for one heavy warship, then a heavy warship's cost spent on interceptors is going to be vastly stronger than either, and that'd be a problem.

I feel like an interceptor should be able to take out a heavy, but only when luck and incredible skill are both involved.

A player as an interceptor pilot could perhaps be able to out-class an enemy heavy (same tier), provided your ship is customized (not "just another jav build"). And with skill, luck, and knowledge, take one out.

**Comment by Quantumshark** [comment]
Define 'incredible skill'. Maybe it's already possible - I don't have any way to distinguish between something that's impossible and something that's possible, but requires more skill than I have.

**Comment by Zitchas** [comment]
> I feel like an interceptor should be able to take out a heavy, but only when luck and incredible skill are both involved.

The difference between an interceptor taking on a heavy and a heavy taking on a heavy is that for the interceptor it requires both a highly specific high-reward-high-risk loadout, a lot of skill, and probably a bit of luck too, while the heavy just needs to out-gun, outlast, and/or outskill the opposing ship. Think torpedo bombers from WWII vs a battleship. The interceptor will be armed with the highest damage weapon they can get, set up to maximize the chance of delivery. The cost, of course, is endurance. The interceptor will probably get one shot to win, and at a fairly high risk to themselves. If they miss, then that's it, they're done. Another heavy, on the other hand, can take a huge amount of punishment, and can afford to miss a fair bit. It's a more consistent and reliable win, but probably not as fast. In effect, I'm talking about a "purpose built interceptor specifically built to defeat that specific configuration of heavy" sort of situation. General purpose generic interceptor? Nope, not a chance. 

So from my perspective, if we're talking 1v1 interceptor vs heavy, sure, definitely should be possible. If we're talking 1 vs fleet where an interceptor, in the course of a single continuous battle without opportunity to reload, methodically whittles their way through an entire fleet of heavies and escorts? Nope.  Accomplishing that should be in the realm of legends, something that might be possible for the best of the best to do, but even they need witnesses for it to be believed. 

(As a caveat, I could see it being possible with some super advanced alien weapon that regenerates a super-deadly torpedo or heavy rocket every few minutes as a theoretical, but that's more like a late-game Quarg special weapon sort of thing, not something seen or used in the regular campaign)

Basically, thinking that a WWII torpedo bomber vs ship situation is good; but trying to finagle a situation where that single torpedo bomber can take out this cruiser then that carrier then the next cruiser.... Nope. 

And recognizing that they have limits, too. This isn't saying that a highly effective bomber can take out *any* ship, just potentially take out a generic heavy. There is a pretty big difference between a top end interceptor taking out a low-end heavy and taking out a high end heavy. At the high end, a well built heavy should be pretty much immune to anything an interceptor can dish out, period. No matter what their loadout or skill is.

**Comment by Saturnalia9** [comment]
> > I feel like an interceptor should be able to take out a heavy, but only when luck and incredible skill are both involved.
> 
> The difference between an interceptor taking on a heavy and a heavy taking on a heavy is that for the interceptor it requires both a highly specific high-reward-high-risk loadout, a lot of skill, and probably a bit of luck too, while the heavy just needs to out-gun, outlast, and/or outskill the opposing ship. Think torpedo bombers from WWII vs a battleship. The interceptor will be armed with the highest damage weapon they can get, set up to maximize the chance of delivery. The cost, of course, is endurance. The interceptor will probably get one shot to win, and at a fairly high risk to themselves. If they miss, then that's it, they're done. Another heavy, on the other hand, can take a huge amount of punishment, and can afford to miss a fair bit. It's a more consistent and reliable win, but probably not as fast. In effect, I'm talking about a "purpose built interceptor specifically built to defeat that specific configuration of heavy" sort of situation. General purpose generic interceptor? Nope, not a chance. 
> 
> So from my perspective, if we're talking 1v1 interceptor vs heavy, sure, definitely should be possible.

Zitchas got the meaning for me. 😂

**Comment by Saturnalia9** [comment]
I'm not skilled with aggressive small ships. 😅 Mostly in offensive and defensive tanks.

**Comment by Quantumshark** [comment]
<details>
<summary>Some Interceptor vs Heavy Warship 1v1s:</summary>
<br>

Berserker vs Vanguard, won by outmaneuvering and avoiding guns
image
image

Hawk vs Protector, won by kiting with the particle cannon after the Protector expended all its missiles (meteors weren't necessary or useful)
image
image

Fury vs Falcon, won by firing missiles from outside the Falcon's range
image
image

Hawk vs Dreadnought, won by kiting with the particle cannon after the Dreadnought expended all its torpedoes
image
image

Fury vs Leviathan, won by firing missiles from outside the Leviathan's range
image
image

</details>

So, it's doable, and possibly too easy.

**Comment by Saturnalia9** [comment]
🤔 What setup did the AI HWs have?

**Comment by Quantumshark** [comment]
> 🤔 What setup did the AI HWs have?

Stock loadouts in each case. In most cases, there aren't any variants that are massively better equipped than the stock variant, so this is a reasonably good approximation for what a player-owned interceptor vs npc heavy warship interaction is usually going to look like.

> I'd like to say "HWs are good at killing other HWs and some HW-imitator MWs. Interceptors are good at kiting and killing slow unprotected HWs. This is what a balanced fleet is for."

HWs are going to perform, on average, averagely against other HWs. That goes for any class - you can't be at an advantage in a mirror match.
More generally though, I do not think "X class is good at beating Y class and bad against Z class" is the ideal model for inter-category balance. I'd much rather something more along the lines of "X class' stats are broadly well-suited to Y _task_, but make Z _task_ particularly challenging **(albeit still possible with a clever approach)** (but individual ships may deviate considerably from others in the same category)."
Also, a 'balanced fleet' shouldn't be considered to be a requirement. A fleet shouldn't be considered to be a requirement at all, balanced or otherwise.

> But at the same time, stock HWs don't realistically do as much damage as they could because AM turrets usually take up several turret mounts (and sometimes take up all the mounts). Meanwhile, player HWs do deal that damage we expect to come from HWs, in some way or another (usually alien tech, though).

Lack of damage output certainly wasn't a factor here. But yes, stock loadouts, and other NPC-controlled variants, tend not to be perfectly optimised - which is a good thing, because it means there's room for the player to assemble a better build and gain an advantage by doing so.

**Comment by Saturnalia9** [comment]
> HWs are going to perform, on average, averagely against other HWs. That goes for any class - you can't be at an advantage in a mirror match.

I assume they would be designed to be able to 1v1 each other with no major advantages/disadvantages.

**Comment by warp-core** [comment]
> <details> <summary>Some Interceptor vs Heavy Warship 1v1s:</summary>
> 
> Berserker vs Vanguard, won by outmaneuvering and avoiding guns image image
> 
> Hawk vs Protector, won by kiting with the particle cannon after the Protector expended all its missiles (meteors weren't necessary or useful) image image
> 
> Fury vs Falcon, won by firing missiles from outside the Falcon's range image image
> 
> Hawk vs Dreadnought, won by kiting with the particle cannon after the Dreadnought expended all its torpedoes image image
> 
> Fury vs Leviathan, won by firing missiles from outside the Leviathan's range image image
> 
> Note that the screenshots of my loadout were taken after the fight; in each case I started with a full load of missiles.
> 
> </details>
> 
> So, it's doable, and possibly too easy.

Did you try any of this stuff in the game without the changes from this PR?

**Comment by Quantumshark** [comment]
> Did you try any of this stuff in the game without the changes from this PR?

Most of it. The Berserker vs Vanguard I couldn't win without these changes, the ones that were won by kiting with a particle cannon were still doable without the changes, but were much harder to get right and took me several attempts. I haven't tried the setups with the sidewinder pods.

**Comment by Saturnalia9** [comment]
Think HWs could add a little hull and/or shield in a separate PR now that they feel heavier? Also wondering what increased the ease for an interceptor to be able to solo a HW... Unless it's mostly the sheer lack of everything in stock HW vs player-optimized interceptor (and the hit to turning).

I still think HWs, for the most part, should have more turrets in exchange for gun ports. And with the mobility nerf, that might be more accepted if the PR goes through.

**Comment by Quantumshark** [comment]
No, I'm not planning on redesigning the whole game from the ground up. If the feeling is that heavy warships would need an HP buff and/or adjustments to their armaments with these changes, then that would indicate that these changes go further than is necessary, and could be dialled back a bit.

**Comment by Saturnalia9** [comment]
> "...Then that would indicate that these changes go further than is necessary, and could be dialled back a bit."

Honestly,I think it's fine as is for the time being. The PR has gotten overall positive feedback, and it's a player ship that beats an AI ship. It'll almost always be one-sided by default.

**Comment by Quantumshark** [comment]
I think that's all the heat dissipations adjusted.

**Comment by Quantumshark** [comment]
A lot of the Syndicate battles in Reconciliation have become much harder with these changes than before, as the Syndicate fleets primarily used smaller ships than the player's FW allies. In some cases they were easy enough before that there's no issue, in others some adjustments are needed.
In general, anywhere where one side consisted of a smaller number of larger ships while the other side was a larger number of smaller ships is going to strongly favour the smaller ships now.

**Comment by Quantumshark** [comment]
Honestly, I think these changes are an overadjustment. Large groups of small ships are just too strong now, both in terms of performing too well compared to other options when deployed by the player, and in terms of being too difficult to deal with when employed by hostile NPC fleets. The latter point could be resolved by adjusting missions and natural spawns, the former could not.
In a similarly problematic vein, Heavy Warships are, I think, nerfed too much here. I agree they shouldn't be the best option for everything, but with these changes they aren't the best option for _anything_. I haven't yet encountered any situation where I felt I'd actually be better off with a Heavy Warship than a smaller ship.
So, this isn't more balanced than the game is without these changes, it's just imbalanced in the opposite direction. At some point soon, I will have a look at opening a new PR with a scaled-back version of these changes, in the hope of closing in on an ideally-balanced experience. When I open that PR, I'll probably close this one.

**Comment by ravenshining** [comment]
I am glad to hear you say that.  I had been thinking the same thing, and wanting to suggest more factors be taken into account -- alien ships might have different scaling or priorities between mass and drag, accounting for chase armament, etc. -- but thought it might be rude to make you redo everything or redo everything for you, not that I have the time to do the latter.

Somewhere on my big computer I thnk I've got data for ship *sprite size* and estimated volume.  I'll pass that along next time I fire it up, if that would help in your mass calculations.  Spite filled area is pretty simple to get, but getting precise model volume from Blender is... possible, but very tedious,  IIRC the plugin to calculate it doesn't take intersections and modifiers into account, and I would not be surprised if it doesn't do scale, either.

**Comment by Quantumshark** [comment]
> I am glad to hear you say that. I had been thinking the same thing, and wanting to suggest more factors be taken into account -- alien ships might have different scaling or priorities between mass and drag, accounting for chase armament, etc. -- but thought it might be rude to make you redo everything or redo everything for you, not that I have the time to do the latter.

I'm always interested to hear more suggestions. This is something that the community have been working on for years, it's probably going to take multiple peoples' ideas to finally come up with a good solution.

My starting point for a second attempt would be to simply take this concept, and scale it down a little - so, a 1.5x multiplier for engines rather than 2x, and scaling back the mass changes such that the increase is halved - i.e. 2x -> 1.5x, 3x -> 2x, 1.8x -> 1.4x etc. So, the same concept, but just going less far, so as to end up somewhere between the current state of the game and where
this PR ended up.

**Comment by Quantumshark** [comment]
Closing in favour of #9616.
